inkcpp/output.cpp

```diff
@@ -5,12 +5,47 @@
 #include <iomanip>
 #endif
 
+#include <cstdio>
+#include <errno.h>
+#include <type_traits>
+
 namespace ink
 {
 	namespace runtime
 	{
 		namespace internal
 		{
+			template<typename T>
+			int toStr(char * buffer, size_t size, T value) {
+				static_assert(!std::is_same<T,T>::value, "Type not supported for conversion!");
+				return EINVAL;
+			}
+
+			// error behavior from: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/itoa-s-itow-s?view=msvc-160
+			template<>
+			int toStr(char * buffer, size_t size, int value) {
+#ifdef WIN32
+				return _itoa_s(value, buffer, size, 10);
+#else
+				if ( buffer == nullptr || size < 1 ){ return EINVAL; }
+				int res = snprintf(buffer, size, "%d", value);
+				if (res > 0 && res < size) { return 0; }
+				return EINVAL;
+#endif
+			}
+
+			// error behavior from: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/gcvt-s?view=msvc-160
+			template<>
+			int toStr(char * buffer, size_t size, float value) {
+#ifdef WIN32
+				return _gcvt_s(buffer, size, value, 11);
+#else
+				if ( buffer == nullptr || size < 1 ) { return EINVAL; }
+				int res = snprintf(buffer, size, "%f.10", value);
+				if (res > 0 && res < size) { return 0; }
+				return EINVAL;
+#endif
+			}
```

Comment:
Maybe this should be written with other platform dependencies in a separate file.

Reply:
I actually have a task to get rid of as many dependencies as I can so I'll decide where all these should go later :)